ipatests/test_integration/test_uninstallation.py

```diff
@@ -21,10 +21,29 @@
 
 class TestUninstallBase(IntegrationTest):
 
+    num_replicas = 1
+
     @classmethod
     def install(cls, mh):
         tasks.install_master(cls.master, setup_dns=False)
 
+    def test_uninstall_client_invalid_hostname(self):
+
+        # using replica as client just for convenience
+        client = self.replicas[0]
+        client_inv_hostname = '{}.nonexistent'.format(client.hostname)
+        tasks.install_client(self.master, client,
+                             extra_args=['--hostname', client_inv_hostname])
+
+        client.run_command(['ipa-client-install', '--uninstall', '-U'])
+        client_uninstall_log = client.get_file_contents(
+            paths.IPACLIENT_UNINSTALL_LOG, encoding='utf-8'
+        )
+        assert "exception: ScriptError:" not in client_uninstall_log
+
+        res_dir = client.run_command(['ls', paths.IPA_CLIENT_SYSRESTORE])
+        assert len(res_dir.stdout_text) == 0
+
```

Comment:
Ah ok so sysrestore.state gets removed when it is empty! This test is a bit implicit since there is no stdout when a file doesn't exist.

I think the test would be clearer if you either looked at stderr for "No such file or directory" or used something like stat to accomplish the same thing.